### DISCUSSION TITLE
Support lenses

### DISCUSSION THREAD

**Discussion by ghost** [mitsuba-renderer/mitsuba3#1032] [Feature requests] [answered]
Is there any easy way to make lenses in mitsuba?

Is it possible to create something like ideal lenses (with focal point on both side and a fixed aperture)?

Or is there any easy way to construct meshes (which might be inaccurate) or other alternatives for evey piece of lenses?

**Comment by sebastienfricker** [comment, marked as answer]
Hi nick008a,
I don't know about ideal lenses. 
You can create a physical lens using mesh shapes for the front, back and edge surfaces. Then set the bsdf to dielectric and you'll get the refraction effect. Just be careful about properly defining the inside / outside refractive indices for each surface (see correctness considerations in the BSDF plugin documentation)
